Sources/System/FilePath/FilePathString.swift

```
@@ -20,6 +20,30 @@ extension FilePath {
     self.init(_platformString: platformString)
   }
 
+  public init(platformString: [CInterop.PlatformChar]) {
+    guard let _ = platformString.firstIndex(of: 0) else {
+      preconditionFailure(
+        "input of FilePath.init(platformString:) must be null-terminated"
```

If this is meant to take a collection of the contents, should we fail if there is an unexpected interior null?

---

My thought is that this should act the same as if it came as a pointer from C, and stop at the first null -- no matter where it is. It is _possible_ that there is code out there that relies on that behaviour.

---

What's the initializer used for? We'd at the very least want to put null-terminated in the argument label.

---

It's a mitigation for an array-to-pointer conversion that leads to reading out of bounds if the array does not contain a null. It has to be an exact overload for `init(platformString: UnsafePointer<CInterop.PlatformChar>)` in order to do its job.